scripts/node_cleanup.py

```diff
@@ -13,13 +13,24 @@ def main():
     stale = query.find_stale_locks(args.owner)
     if not stale:
         return
+    by_owner = {}
+    for node in stale:
+        if args.owner and node['locked_by'] != args.owner:
+            log.warning(
+                f"Node {node['name']} expected to be locked by {args.owner} "
+                f"but found {node['locked_by']} instead"
+            )
+            continue
+        owner_nodes = by_owner.setdefault(node['locked_by'], [])
+        owner_nodes.append(node)
```

Comment:
I know that owner_nodes is a reference to by_owner, can't we just purely use by_owner? Am I missing something?

Reply:
`by_owner` is a dict where keys are e.g. "scheduled_zack@teuthology", and values are lists of dicts containing node data, whereas `owner_nodes` is a reference to one of those lists - but you're right that there's no real point in holding this reference.

I'll replace the above two lines with:
```
by_owner.setdefault(node['locked_by'], []).append(node)
```